Improved monitoring alerts to alerts@zap.me

eoliveros opened the pull request:
Added inventory hosts file, scripts {zapd_balance,zapd_block_height,zapd_test}.sh in templates, and the zapd_cron.yml playbook

eoliveros committed:
Created new branch

Added inventory hosts file, scripts in templates, and the zapd_cron.yml playbook

djpnewton commented on ansible/templates/zapd_balance.sh:
```diff
@@ -0,0 +1,18 @@
+
+
+bal=`curl -s -d '{"jsonrpc":"2.0","id":1,"method":"getbalance","params":{}}' -H "Content-Type: application/json-rpc" localhost:5000/api | jq '.["result"]["balance"]'`
+min_bal=500000
```
lets make constants min_bal 5000 and max_bal 10000, then when you use them here multiply by  a constant CENTS (which will be 100 obviously), this way you can reuse the min/max constants in the email message

djpnewton commented on ansible/templates/zapd_balance.sh:
```diff
@@ -0,0 +1,18 @@
+bal=`curl -s -d '{"jsonrpc":"2.0","id":1,"method":"getbalance","params":{}}' -H "Content-Type: application/json-rpc" localhost:5000/api | jq '.["result"]["balance"]'`
+min_bal=500000
+max_bal=1000000
+dest_email=alerts@zap.me
```
this should come from from a ansible variable

djpnewton commented on ansible/templates/zapd_block_height.sh:
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+dest_email=alerts@zap.me
```
this should come from from an ansible variable

djpnewton commented on ansible/templates/zapd_block_height.sh:
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+dest_email=alerts@zap.me
+remote_block=`curl -s https://testnode1.wavesnodes.com/blocks/height | jq '.["height"]'` 
```
remote url (testnode1.wavesnodes.com) will be different for mainnet, make this come from a ansible variable

djpnewton commented on ansible/templates/zapd_block_height.sh:
```diff
@@ -0,0 +1,16 @@
+max_height_diff=2
+
+### Remote block would be same/higher than local block.
+num=$(( $remote_block - $local_block ))
```
its actually possible for us to get ahead of the specific remote node we are talking to because its a distributed network.. so make this the absolute value of the difference

djpnewton commented on ansible/templates/zapd_test.sh:
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+dest_email="alerts@zap.me"
```
this should come from from an ansible variable

djpnewton commented:
Addresses ZAPD-1

eoliveros committed:
Changes to script.

Changed certain values in the scripts

djpnewton commented on ansible/zapd_cron.yml:
```diff
@@ -4,6 +4,14 @@
+   max_zap: 10000
+   testnet_node: testnet1.wavesnodes.com
+   mainnet_node: nodes.wavesnodes.com
+
```
waves_node: "{{ 'testnet1.wavesnodes.com' if TESTNET else 'nodes.wavesnodes.com' }}"

we have a variable named TESTNET which is set in ansible_deploy.sh

djpnewton replied:
actually call this variable "remote_waves_node" or something like that

eoliveros replied:
not sure why that variable is still showing in the zapd_cron.yml. I deleted this before the push. This is being pushed from the ansible_deploy.sh script.

djpnewton commented on ansible/templates/zapd_block_height.sh:
```diff
@@ -1,15 +1,32 @@
 
-### Remote block would be same/higher than local block.
-num=$(( $remote_block - $local_block ))
+{% if ansible_hostname == 'testnet' %}
```
see below.. use the TESTNET variable already defined project wide by ansible_deploy.sh

djpnewton commented on ansible/zapd_cron.yml:
```diff
@@ -4,6 +4,14 @@
   become_user: root
 
+  vars:
+   dest_email: alerts@zap.me
```
ansible_deploy.sh sets a var (ADMIN_EMAIL=admin@zap.me) add one also there for alerts@zap.me

eoliveros committed:
Added variable ALERT_EMAIL

 is alerts@zap.me

eoliveros committed:
Changed to include ALERT_EMAIL

include print the  variable

eoliveros committed:
Removed variables

Removed variables for dest_email and 2 others.

eoliveros committed:
Uncommented the playbooks from erick's test.

eoliveros committed:
Changed the dest_email to get the variable from the ansible_deploy.sh script

eoliveros committed:
Changed the dest_email variable to come from the ansible_deploy.sh script

djpnewton commented on ansible/templates/zapd_block_height.sh:
```diff
@@ -0,0 +1,34 @@
+max_height_diff=2
+
+
+{% if DEPLOY_HOST == 'testnet.zap.me' %}
```
just use the "remote_waves_node" here

eoliveros replied:
DEPLOY_HOST is a variable passed from the ansible_deploy.sh script. Do you want me to change the   ansible_deploy.sh script?

if [[ ( $DEPLOY_TYPE == "test" ) ]]
then
    DEPLOY_HOST=testnet.zap.me
    DEPLOY_USER=root
    REMOTE_WAVES_NODE=<name of the node>
    TESTNET=true
fi

Then pass the "REMOTE_WAVES_NODES" to the ansible-playbook?

djpnewton replied:
ok sure

djpnewton commented on ansible/zapd_cron.yml:
```diff
@@ -0,0 +1,71 @@
+        dest: /root/scripts/zapd_balance.sh
+        mode: 0755
+   
+    - name : DEBUG
```
remove this?

djpnewton commented on ansible/zapd_cron.yml:
```diff
@@ -0,0 +1,71 @@
+        dest: /root/scripts/zapd_block_height.sh
+        mode: 0755
+    
+    - name: copy the zapd_test.sh using the template module
```
could we rename this to describe what its testing? zapd_isalive.sh or something?

djpnewton commented:
looking good, almost ready

eoliveros committed:
Changed the script name and cronjob

Changed the name of the script from zapd_test.sh to zapd_heartbeat.sh adn the cron job associated with the script

eoliveros committed:
Changed ansible_deploy.sh to include REMOTE_WAVES_NODES

Added a new variable to pass to the ansible-playbook

eoliveros committed:
Changed the script

Instead of looking at the variable from teh zapd_cron, the variable is set from the ansible_deploy.sh script.

djpnewton committed:
Merge branch 'master' into zapd_cron

djpnewton committed:
moved old cron stuff into zapd_cron.yml
fixed whitespace changes

djpnewton committed:
fix heartbeat script name

djpnewton committed:
fix the crontab scripts location

djpnewton committed:
fix zap_bal division
simplify zap_block_height.sh

djpnewton committed:
add zapd scanned block height to alerts
